Bottom: e4ab7d162f9c5eede46be39fbf11dbee14c2525a
Top:    12e50888352384de5f9ee23f448bf032ab30b794
Author: Jon Smirl <jonsmirl@smirl2.(none)>
Date:   2013-06-21 13:44:58 -0400

Basic changes for AsiaRF module


---

diff --git a/.gitignore b/.gitignore
index 70a454c..eec44d1 100644
--- a/.gitignore
+++ b/.gitignore
@@ -8,11 +8,11 @@
 /bin
 /build_dir
 /staging_dir
+/patches*
 /tmp
 /logs
 /feeds
 /feeds.conf
-/files
 /package/feeds
 /package/openwrt-packages
 *.orig
@@ -21,4 +21,4 @@
 .#*
 *#
 .emacs.desktop*
-TAGS*~
\ No newline at end of file
+TAGS*~
diff --git a/files/etc/config/batman-adv b/files/etc/config/batman-adv
new file mode 100644
index 0000000..361ecdf
--- /dev/null
+++ b/files/etc/config/batman-adv
@@ -0,0 +1,3 @@
+config mesh 'bat0'
+        option 'aggregated_ogms'
+        option 'ap_isolation' 
diff --git a/files/etc/config/dhcp b/files/etc/config/dhcp
new file mode 100644
index 0000000..1db87c6
--- /dev/null
+++ b/files/etc/config/dhcp
@@ -0,0 +1,38 @@
+config dnsmasq
+	option domainneeded	1
+	option boguspriv	1
+	option filterwin2k	0  # enable for dial on demand
+	option localise_queries	1
+	option rebind_protection 1  # disable if upstream must serve RFC1918 addresses
+	option rebind_localhost 1  # enable for RBL checking and similar services
+	#list rebind_domain example.lan  # whitelist RFC1918 responses for domains
+	option local	'/lan/'
+	option domain	'lan'
+	option expandhosts	1
+	option nonegcache	0
+	option authoritative	1
+	option readethers	1
+	option leasefile	'/tmp/dhcp.leases'
+	option resolvfile	'/tmp/resolv.conf.auto'
+	#list server		'/mycompany.local/1.2.3.4'
+	#option nonwildcard	1
+	#list interface		br-lan
+	#list notinterface	lo
+	#list bogusnxdomain     '64.94.110.11'
+
+config dhcp lan
+	option interface	lan
+	option start 	100
+	option limit	150
+	option leasetime	12h
+
+config dhcp hostspot
+        option interface 'hotspot'
+        option start '100'
+        option limit '150'
+        option dynamicdhcp '1'
+
+config dhcp wan
+	option interface	wan
+	option ignore	1
+
diff --git a/files/etc/config/network b/files/etc/config/network
new file mode 100644
index 0000000..a24e7c5
--- /dev/null
+++ b/files/etc/config/network
@@ -0,0 +1,49 @@
+config interface 'loopback'
+option ifname 'lo'
+option proto 'static'
+option ipaddr '127.0.0.1'
+option netmask '255.0.0.0'
+
+config globals 'globals'
+option ula_prefix 'fd28:7772:4f77::/48'
+
+config switch 'eth0'
+option name 'switch0'
+option reset '1'
+        option vlan0 '1 2 3 4 5*'
+        option vlan1 '0 5'
+        option vlan2 '5'
+
+config interface 'lan'
+        option ifname 'eth0.0'
+        option proto 'static'
+        option ipaddr '192.168.2.2'
+        option netmask '255.255.255.0'
+        option defaultroute '0'
+        option peerdns '0'
+        option type 'bridge'
+	option ip6assign '60'
+
+config interface 'wan'
+        option ifname 'eth0.1'
+        option defaultroute '0'
+        option peerdns '0'
+        option proto 'static'
+        option ipaddr '192.168.3.2'
+        option netmask '255.255.255.0'
+        option gateway '192.168.3.1'
+        option dns '192.168.3.1'
+
+config interface 'hotspot'
+        option ifname 'eth0.2'
+        option proto 'static'
+        option ipaddr '192.168.4.2'
+        option netmask '255.255.255.0'
+        option type 'bridge'
+        option defaultroute '0'
+
+config interface 'mesh'
+        option ifname 'adhoc0'
+        option mtu '1528'
+        option proto 'batadv'
+        option mesh 'bat0'
diff --git a/files/etc/config/wireless b/files/etc/config/wireless
new file mode 100644
index 0000000..4cfe7be
--- /dev/null
+++ b/files/etc/config/wireless
@@ -0,0 +1,36 @@
+config wifi-device radio0
+	option type     mac80211
+	option channel  11
+	option hwmode	11ng
+	option path	'10180000.wmac'
+	option htmode	HT20
+	list ht_capab	GF
+	list ht_capab	SHORT-GI-20
+	list ht_capab	SHORT-GI-40
+	list ht_capab	RX-STBC1
+
+config wifi-iface
+        option device 'radio0'
+        option ifname 'ap0'
+        option network 'lan'
+        option mode 'ap'
+        option ssid 'Smirl secure'
+        option encryption 'psk2'
+        option key 'iebgener'
+
+config wifi-iface
+        option device 'radio0'
+        option ifname 'ap1'
+        option ssid 'Smirl open'
+        option network 'hotspot'
+        option mode 'ap'
+        option encryption 'none'
+        option isolate '1'
+
+config wifi-iface
+        option device 'radio0'
+        option ifname 'adhoc0'
+        option network 'mesh'
+        option mode 'adhoc'
+        option ssid 'mesh'
+        option bssid '02:CA:FE:CA:CA:40'
diff --git a/target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom b/target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom
index 810dddf..aace5b5 100644
--- a/target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom
+++ b/target/linux/ramips/base-files/etc/hotplug.d/firmware/10-rt2x00-eeprom
@@ -61,6 +61,7 @@ case "$FIRMWARE" in
 	all5002 | \
 	all5003 | \
 	argus-atp52b | \
+	awm002 | \
 	bc2 | \
 	br6425 | \
 	broadway | \
diff --git a/target/linux/ramips/base-files/etc/uci-defaults/02_network b/target/linux/ramips/base-files/etc/uci-defaults/02_network
index 747a83d..49cc0cd 100755
--- a/target/linux/ramips/base-files/etc/uci-defaults/02_network
+++ b/target/linux/ramips/base-files/etc/uci-defaults/02_network
@@ -41,6 +41,7 @@ ramips_setup_interfaces()
 	all0256n | \
 	all5002 | \
 	all5003 | \
+	awm002 | \
 	broadway | \
 	wnce2001)
 		ucidef_add_switch "switch0" "1" "0"
diff --git a/target/linux/ramips/base-files/lib/preinit/06_set_iface_mac b/target/linux/ramips/base-files/lib/preinit/06_set_iface_mac
index bd1dac0..a612b75 100644
--- a/target/linux/ramips/base-files/lib/preinit/06_set_iface_mac
+++ b/target/linux/ramips/base-files/lib/preinit/06_set_iface_mac
@@ -55,6 +55,7 @@ preinit_set_mac_address() {
 	all0256n |\
 	all5002 |\
 	all5003 |\
+	awm002 |\
 	carambola |\
 	dir-615-h1 |\
 	fonera20n |\
diff --git a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
index c5f216f..d9ee907 100755
--- a/target/linux/ramips/base-files/lib/ramips.sh
+++ b/target/linux/ramips/base-files/lib/ramips.sh
@@ -46,6 +46,9 @@ ramips_board_detect() {
 	*"Argus ATP-52B")
 		name="argus-atp52b"
 		;;
+	*"AsiaRF AWM002")
+		name="awm002"
+		;;
 	*"BR6524N")
 		name="br6524n"
 		;;
diff --git a/target/linux/ramips/base-files/lib/upgrade/platform.sh b/target/linux/ramips/base-files/lib/upgrade/platform.sh
index eaa7e64..a15d906 100755
--- a/target/linux/ramips/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ramips/base-files/lib/upgrade/platform.sh
@@ -24,6 +24,7 @@ platform_check_image() {
 	all5002 | \
 	all5003 | \
 	asl26555 | \
+	awm002 | \
 	bc2 | \
 	broadway | \
 	carambola | \
diff --git a/target/linux/ramips/dts/AWM002-4M.dts b/target/linux/ramips/dts/AWM002-4M.dts
new file mode 100644
index 0000000..adadfa1
--- /dev/null
+++ b/target/linux/ramips/dts/AWM002-4M.dts
@@ -0,0 +1,108 @@
+/dts-v1/;
+
+/include/ "rt5350.dtsi"
+
+/ {
+	compatible = "AWM002", "ralink,rt5350-soc";
+	model = "AsiaRF AWM002";
+
+	palmbus@10000000 {
+		sysc@0 {
+			ralink,pinmux = "spi", "uartlite", "jtag", "mdio", "sdram", "rgmii";
+			ralink,gpiomux = "i2c";
+			ralink,uartmux = "gpio";
+			ralink,wdtmux = <1>;
+		};
+
+		gpio0: gpio@600 {
+			status = "okay";
+		};
+
+		spi@b00 {
+			status = "okay";
+			m25p80@0 {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "mx25l6405d";
+				reg = <0 0>;
+				linux,modalias = "m25p80", "mx25l6405d";
+				spi-max-frequency = <10000000>;
+
+		partition@0 {
+			label = "u-boot";
+			reg = <0x0 0x30000>;
+			read-only;
+		};
+
+		partition@30000 {
+			label = "u-boot-env";
+			reg = <0x30000 0x10000>;
+			read-only;
+		};
+
+		factory: partition@40000 {
+			label = "factory";
+			reg = <0x40000 0x10000>;
+			read-only;
+		};
+
+		partition@50000 {
+			label = "firmware";
+			reg = <0x50000 0x1fb0000>;
+		};
+			};
+		};
+	};
+
+	ethernet@10100000 {
+		status = "okay";
+	};
+
+	esw@10110000 {
+		status = "okay";
+		ralink,portmap = <0x3f>;
+	};
+
+	wmac@10180000 {
+		status = "okay";
+	};
+
+	ehci@101c0000 {
+		status = "okay";
+	};
+
+	ohci@101c1000 {
+		status = "okay";
+	};
+
+	i2c-gpio {
+		compatible = "i2c-gpio";
+
+		gpios = <&gpio0 1 0 &gpio0 2 0>;
+		i2c-gpio,delay-us = <10>;
+
+		pcf0: iexp@38 {
+			#gpio-cells = <2>;
+			compatible = "inxp,pcf8574a";
+			reg = <0x38>;
+			gpio-controller;
+		};
+
+		hwmon@4b {
+			compatible = "national,lm92";
+			reg = <0x4b>;
+		};
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+		ld1 {
+			label = "asiarf:green:ld1";
+			gpios = <&pcf0 0 1>;
+		};
+		ld2 {
+			label = "asiarf:green:ld2";
+			gpios = <&pcf0 1 1>;
+		};
+	};
+};
diff --git a/target/linux/ramips/dts/AWM002-8M.dts b/target/linux/ramips/dts/AWM002-8M.dts
new file mode 100644
index 0000000..b112981
--- /dev/null
+++ b/target/linux/ramips/dts/AWM002-8M.dts
@@ -0,0 +1,89 @@
+/dts-v1/;
+
+/include/ "rt5350.dtsi"
+
+/ {
+	compatible = "AWM002", "ralink,rt5350-soc";
+	model = "AsiaRF AWM002";
+
+	palmbus@10000000 {
+		sysc@0 {
+			ralink,pinmux = "spi", "uartlite", "jtag", "mdio", "sdram", "rgmii";
+			ralink,gpiomux = "i2c";
+			ralink,uartmux = "gpio";
+			ralink,wdtmux = <1>;
+		};
+
+		gpio0: gpio@600 {
+			status = "okay";
+		};
+
+		spi@b00 {
+			status = "okay";
+			m25p80@0 {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "mx25l6405d";
+				reg = <0 0>;
+				linux,modalias = "m25p80", "mx25l6405d";
+				spi-max-frequency = <10000000>;
+
+				partition@0 {
+					label = "u-boot";
+					reg = <0x0 0x30000>;
+					read-only;
+				};
+
+				partition@30000 {
+					label = "u-boot-env";
+					reg = <0x30000 0x10000>;
+					read-only;
+				};
+
+				factory: partition@40000 {
+					label = "factory";
+					reg = <0x40000 0x10000>;
+					read-only;
+				};
+
+				partition@50000 {
+					label = "firmware";
+					reg = <0x50000 0x1fb0000>;
+				};
+			};
+		};
+	};
+
+	ethernet@10100000 {
+		status = "okay";
+	};
+
+	esw@10110000 {
+		status = "okay";
+		ralink,portmap = <0x3f>;
+	};
+
+	wmac@10180000 {
+		status = "okay";
+	};
+
+	ehci@101c0000 {
+		status = "okay";
+	};
+
+	ohci@101c1000 {
+		status = "okay";
+	};
+
+	gpio-leds {
+		compatible = "gpio-leds";
+		ld1 {
+			label = "asiarf:green:ld1";
+			gpios = <&gpio0 0 1>;
+		};
+		ld2 {
+			label = "asiarf:green:ld2";
+			gpios = <&gpio0 1 1>;
+		};
+	};
+};
diff --git a/target/linux/ramips/image/Makefile b/target/linux/ramips/image/Makefile
index 5f2c311..399a5e7 100644
--- a/target/linux/ramips/image/Makefile
+++ b/target/linux/ramips/image/Makefile
@@ -279,6 +279,8 @@ define BuildFirmware/ASL26555/initramfs
 endef
 Image/Build/Profile/ASL26555=$(call BuildFirmware/ASL26555/$(1),$(1),asl26555,ASL26555)
 
+Image/Build/Profile/AWM002=$(call BuildFirmware/DefaultDualSize/$(1),$(1),awm002,AWM002)
+
 Image/Build/Profile/BC2=$(call BuildFirmware/Default8M/$(1),$(1),bc2,BC2)
 
 broadway_mtd_size=7929856
@@ -444,6 +446,7 @@ define Image/Build/Profile/Default
 	$(call Image/Build/Profile/ALL5003,$(1))
 	$(call Image/Build/Profile/ARGUS_ATP52B,$(1))
 	$(call Image/Build/Profile/ASL26555,$(1))
+	$(call Image/Build/Profile/AWM002,$(1))
 	$(call Image/Build/Profile/BC2,$(1))
 	$(call Image/Build/Profile/BROADWAY,$(1))
 	$(call Image/Build/Profile/CARAMBOLA,$(1))
diff --git a/target/linux/ramips/rt305x/profiles/asiarf.mk b/target/linux/ramips/rt305x/profiles/asiarf.mk
new file mode 100644
index 0000000..7afa253
--- /dev/null
+++ b/target/linux/ramips/rt305x/profiles/asiarf.mk
@@ -0,0 +1,18 @@
+#
+# Copyright (C) 2012 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+define Profile/AWM002
+	NAME:=AsiaRF AWM002
+	PACKAGES:=kmod-usb-core kmod-usb-ohci kmod-usb2 kmod-ledtrig-usbdev \
+	kmod-i2c-core kmod-i2c-gpio
+endef
+
+define Profile/AWM002/Description
+	Package set for AsiaRF AWM002
+endef
+
+$(eval $(call Profile,AWM002))
